core/src/main/java/org/apache/metamodel/DeleteAndInsertBuilder.java

```diff
@@ -69,12 +69,33 @@ public void execute() throws MetaModelException {
     private List<Row> updateRows(List<Row> rows) {
         for (ListIterator<Row> it = rows.listIterator(); it.hasNext();) {
             final Row original = (Row) it.next();
+            validateUpdateType(original);
```

Comment:
removed it from the class with next commit, I think it was to prevent the insertion of for example a String value into a Column which only contains Integers and therefore has the columnType of Integer.